cloud/scope/managed_machine_pool.go

```diff
@@ -605,11 +612,15 @@ func (m *ManagedMachinePoolScope) UpdateNodePool(ctx context.Context, pool *oke.
 func setMachinePoolSpecDefaults(spec *infrav1exp.OCIManagedMachinePoolSpec) {
 	spec.ProviderIDList = nil
 	spec.ProviderID = nil
-	if spec.NodePoolNodeConfig != nil && spec.NodePoolNodeConfig.PlacementConfigs != nil {
-		configs := spec.NodePoolNodeConfig.PlacementConfigs
-		sort.Slice(configs, func(i, j int) bool {
-			return *configs[i].AvailabilityDomain < *configs[j].AvailabilityDomain
-		})
+	if spec.NodePoolNodeConfig != nil {
+		if spec.NodePoolNodeConfig.PlacementConfigs != nil {
+			configs := spec.NodePoolNodeConfig.PlacementConfigs
+			sort.Slice(configs, func(i, j int) bool {
+				return *configs[i].AvailabilityDomain < *configs[j].AvailabilityDomain
+			})
+		}
+		// set to default value as we should not compare this field
+		spec.NodePoolNodeConfig.UpdateNodePoolSize = false
```

Comment:
What do you think about setting this as a default in a default webhook instead?

Reply:
this is creating the spec here for comparison, we cant do it in webhook, as we do this every time we create a spec for comparison.